Landsat import can't recover from missing s3 objects

If the USGS csv is wrong (which is pretty infrequent, but it does happen sometimes), we go searching for a key that doesn't exist. When that happens, the import fails and can't recover. Given that this is a foreseeable failure, we should recover from it.

From 2016-06-07, e.g.:

`17:24:55.724 [ERROR] c.a.r.b.l.a.ImportLandsat8 - com.azavea.shaded.amazonaws.services.s3.model.AmazonS3Exception: The specified key does not exist. (Service: Amazon S3; Status Code: 404; Error Code: NoSuchKey; Request ID: 4BD005570CF6A684), S3 Extended Request ID: aPUPwnpx/Kt1VgenVNS9ccsH0DGCTFvYCX2kPMktvzOkrYmPUpbofjxDV1hZXTvodxkntq3Gmt0=`

--- a/app-backend/batch/src/main/scala/com/azavea/rf/batch/landsat8/airflow/ImportLandsat8.scala
+++ b/app-backend/batch/src/main/scala/com/azavea/rf/batch/landsat8/airflow/ImportLandsat8.scala
@@ -19,7 +19,7 @@ import java.util.UUID
 
 import scala.collection.mutable.ListBuffer
 import scala.concurrent.Future
-import scala.util.{Failure, Success}
+import scala.util.{Try, Failure, Success}
 import scala.util.control.Breaks._
 
 @deprecated(message = "Use com.azavea.rf.batch.landsat8.airflow.ImportLandsat8C1 instead, as USGS changes LC8 storage rules: https://landsat.usgs.gov/landsat-collections", since = "https://github.com/azavea/raster-foundry/pull/1821")
@@ -74,10 +74,18 @@ case class ImportLandsat8(startDate: LocalDate = LocalDate.now(ZoneOffset.UTC), 
     rootUrl
   }
 
+  // Getting the image size is the only place where the s3 object
+  // is required to exist -- so handle the missing object by returning
+  // a -1 for the image's size
   protected def sizeFromPath(tifPath: String, productId: String): Int = {
     val path = s"${getLandsatPath(productId)}/$tifPath"
     logger.info(s"Getting object size for path: $path")
-    s3Client.getObject(landsat8Config.bucketName, path).getObjectMetadata.getContentLength.toInt
+    Try(
+      s3Client.getObject(landsat8Config.bucketName, path).getObjectMetadata.getContentLength.toInt
+    ) match {
+      case Success(size) => size
+      case Failure(_) => -1
+    }
   }
 
   protected def createThumbnails(sceneId: UUID, productId: String): List[Thumbnail.Identified] = {

--- a/app-backend/batch/src/main/scala/com/azavea/rf/batch/landsat8/airflow/ImportLandsat8C1.scala
+++ b/app-backend/batch/src/main/scala/com/azavea/rf/batch/landsat8/airflow/ImportLandsat8C1.scala
@@ -19,7 +19,7 @@ import java.util.UUID
 
 import scala.collection.mutable.ListBuffer
 import scala.concurrent.Future
-import scala.util.{Failure, Success}
+import scala.util.{Try, Failure, Success}
 import scala.util.control.Breaks._
 
 case class ImportLandsat8C1(startDate: LocalDate = LocalDate.now(ZoneOffset.UTC), threshold: Int = 10)(implicit val database: DB) extends Job {
@@ -74,10 +74,18 @@ case class ImportLandsat8C1(startDate: LocalDate = LocalDate.now(ZoneOffset.UTC)
     rootUrl
   }
 
+  // Getting the image size is the only place where the s3 object
+  // is required to exist -- so handle the missing object by returning
+  // a -1 for the image's size
   protected def sizeFromPath(tifPath: String, productId: String): Int = {
     val path = s"c1/${getLandsatPath(productId)}/$tifPath"
     logger.info(s"Getting object size for path: $path")
-    s3Client.getObject(landsat8Config.bucketName, path).getObjectMetadata.getContentLength.toInt
+    Try(
+      s3Client.getObject(landsat8Config.bucketName, path).getObjectMetadata.getContentLength.toInt
+    ) match {
+      case Success(size) => size
+      case Failure(_) => -1
+    }
   }
 
   protected def createThumbnails(sceneId: UUID, productId: String): List[Thumbnail.Identified] = {

